airflow/contrib/operators/gcp_tasks_operator.py

```diff
@@ -0,0 +1,939 @@
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from airflow.contrib.hooks.gcp_tasks_hook import CloudTasksHook
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+
+
+class CloudTasksCreateQueueOperator(BaseOperator):
+    """
+    Creates a queue in Cloud Tasks.
+
+    :param location: The location name in which the queue will be created.
+    :type location: str
+    :param task_queue: The task queue to create.
+        Queue's name cannot be the same as an existing queue.
+        If a dict is provided, it must be of the same form as the protobuf message Queue.
+    :type task_queue: dict or class google.cloud.tasks_v2.types.Queue
+    :param project_id: (Optional) The ID of the  GCP project that owns the Cloud Tasks.
+        If set to None or missing, the default project_id from the GCP connection is used.
+    :type project_id: str
+    :param queue_name: (Optional) The queue's name.
+        If provided, it will be used to construct the full queue path.
+    :type queue_name: str
+    :param retry: (Optional) A retry object used to retry requests.
+        If None is specified, requests will not be retried.
+    :type retry: google.api_core.retry.Retry
+    :param timeout: (Optional) The amount of time, in seconds, to wait for the request
+        to complete. Note that if retry is specified, the timeout applies to each
+        individual attempt.
+    :type timeout: float
+    :param metadata: (Optional) Additional metadata that is provided to the method.
+    :type metadata: sequence[tuple[str, str]]]
+    :param gcp_conn_id: (Optional) The connection ID used to connect to Google Cloud Platform.
+    :type gcp_conn_id: str
+    :rtype: google.cloud.tasks_v2.types.Queue
+    """
+
+    template_fields = (
+        "task_queue",
+        "project_id",
+        "location",
+        "queue_name",
+        "gcp_conn_id",
+    )
+
+    @apply_defaults
+    def __init__(
+        self,
+        location,
+        task_queue,
+        project_id=None,
+        queue_name=None,
+        retry=None,
+        timeout=None,
+        metadata=None,
+        gcp_conn_id="google_cloud_default",
+        *args,
+        **kwargs
+    ):
+        super().__init__(*args, **kwargs)
+        self.location = location
+        self.task_queue = task_queue
+        self.project_id = project_id
+        self.queue_name = queue_name
+        self.retry = retry
+        self.timeout = timeout
+        self.metadata = metadata
+        self.gcp_conn_id = gcp_conn_id
+        self._hook = CloudTasksHook(gcp_conn_id=self.gcp_conn_id)
```

Comment:
I think, you should create hook in `execute`. Otherwise, templates will not work.